src/rust/ide/view/graph-editor/src/component/node.rs

```diff
@@ -562,25 +563,33 @@ impl Node {
             eval out.hover ((t) action_bar.set_visibility(t));
 
 
-            // === Set Node Error ===
+            // === Errors on Node ===
+
+            is_error_set <- frp.error.map(|err| err.is_some());
 
-            error_color_anim.target <+ frp.set_error.map(f!((error) {
+            frp.source.error <+ frp.set_error.map(f!([model](error) {
                 model.set_error(error.as_ref());
-                match error.as_ref() {
-                    Some(error) => {
-                        let error_data:visualization::Data = error.clone().into();
-                        model.visualization.frp.set_data.emit(error_data);
-                        color::Lcha::from(color::Rgba::red())
-                    },
-                    None => color::Lcha::transparent(),
-                }
+                error.clone()
             }));
 
+            error_color_anim.target <+ frp.error.map(f!([style](error)
+                Self::error_color(error,&style))
+            );
+
             eval error_color_anim.value((value)
                 model.error_indicator.color_rgba.set(color::Rgba::from(value).into());
             );
 
 
+            // === Visualization ===
+
+            eval frp.set_visualization ((t) model.visualization.frp.set_visualization.emit(t));
+            visualization_enabled <- bool(&frp.disable_visualization,&frp.enable_visualization);
+            visualization_visible <- visualization_enabled.all_with(&is_error_set, |l,r| *l && !*r);
```

Comment:
use `visualization_visible <- visualization_enabled && ! is_error_set;` instead :) I'm not sure whether we support `!` in macros, but if we don't, add it there please. We support `&&` for sure.